symengine/basic-inl.h

```diff
@@ -14,12 +14,15 @@ inline hash_t Basic::hash() const
 //! \return true if not equal
 inline bool Basic::__neq__(const Basic &o) const
 {
-    return not(this->__eq__(o));
+    return not(eq(*this, o));
```

Comment:
I called `__eq__` probably because I was trying to save a function call. But it's in a header file, so probably no overhead there, and now there is the extra optimization in `eq`.

Reply:
Yeah, this is inlined.